### ISSUE TITLE
Changing the size

### ISSUE THREAD

**Issue by arturochian** [armish/coffeewheel#1]
Hi  @armish this is a really awesome package and a nice example of html widget.

Are there some way to change the size of each part? something like this image

http://prntscr.com/5nd32u  (http://bl.ocks.org/mbostock/4348373)

PD: Happy new year!

**Comment by armish** [comment]
Hey Arturo,

Right now, the partition function that decides on the sizes of the rims takes only the number of `children` each node has into account. Partitioning based on an arbitrary value associated with the nodes should be easy to implement in a way that user passes it to the plotting function as a string parameter. I will do it (and update this entry accordingly), but probably won't be able to get to it until Monday.

Thanks for your comment, I am glad you liked it.

Happy new year,
-- Arman

**Comment by arturochian** [comment]
Hi @armish,

Is Partition function on this package? 

PD: Do you know if there is a fast way (maybe a package or stackoverflow solution you know) to convert dataframe to list (coffeewheel structure) in order to create this widget?

**Comment by armish** [comment]
Hey @arturochian,

No, the partition function is not in the package, but is included in the Javascript code. AFAIK, there is no \* de facto\* function to convert the dataframe into a nested list; you might need to iterate over the dataframe to create the structure :(

**Comment by happyshows** [comment]
@armish will there be any update on this package regarding the size of each slice or other features?

**Comment by armish** [comment]
Hi @happyshows

Yes, that's definitely the plan; let me get it done by the end of this week. I will update this issue as soon as it is done.

Thanks,
-- Arman

**Comment by happyshows** [comment]
@armish just checking, any progress on this issue?

**Comment by armish** [comment]
Sorry that this took so long, but you should now be able to use any attribute to define the partition. All you need is to add an attribute to the leaf nodes that represents the partition size and the rest will be taken care of. By default, the partition is done on the "value" attribute and if that does not exist, then everything is evenly partitioned.

Hope this helps,
-- Arman

**Comment by arturochian** [comment]
Thank you @armish. 

PD: Could you provide an example using this parameter?

**Comment by armish** [comment]
Hi @arturochian,

Sure; for example the following:

``` r
wheelData <- list(
    list(name="10percent", value=10, colour="red"),
    list(name="30percent", value=30, colour="green"),
    list(name="60percent", value=60, colour="blue")
);

coffeewheel(treeData=wheelData);
```

will now give you this plot:
![screen shot 2015-04-29 at 2 36 08 pm](https://cloud.githubusercontent.com/assets/7809/7398473/450f7890-ee7d-11e4-8264-92d3e0ac21e7.png)

Hope this helps,
-- Arman

**Comment by happyshows** [comment]
@armish thanks for the update Arman. Could you please confirm whether this would work with htmlwidgets 0.3.3?

**Comment by armish** [comment]
@happyshows just upgraded my htmlwidgets to version 0.3.2 (0.3.3 was not available yet) and confirmed that this widget works with the latest version:

```
R> sessionInfo()
R version 3.1.2 (2014-10-31)
Platform: x86_64-apple-darwin10.8.0 (64-bit)

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] coffeewheel_0.1   htmlwidgets_0.3.2

loaded via a namespace (and not attached):
[1] digest_0.6.8    htmltools_0.2.6 RJSONIO_1.3-0   tools_3.1.2     yaml_2.1.13 
```

**Comment by happyshows** [comment]
@armish I believe the latest version of htmlwidget is 0.3.3 on github. I downloaded a few days back and realized the switch broke coffeewheel. The major change is they switched json package. 

https://github.com/ramnathv/htmlwidgets/pull/28
https://github.com/ramnathv/htmlwidgets/issues/101

**Comment by armish** [comment]
Thanks @happyshows; didn't know about this change. Let me create a new issue for this and get back to it as soon as possible. I think I should synchronize this fix with the official release of 0.3.3 to not break things for people who are using the older versions for the time being.

**Comment by happyshows** [comment]
@armish thanks. It's really a breaking upgrade. I uses a couple htmlwidget based packages in one app and would need to either upgrade all packages or not upgrade at all :S  If possible, would you please document the steps to fix it so that I can refer and override in case some package does not provide a new release and I'll try patching by myself.
